modules/tinymce/src/core/main/ts/delete/MergeBlocks.ts

```diff
@@ -55,9 +55,14 @@ const nestedBlockMerge = (
 
 const sidelongBlockMerge = (rootNode: SugarElement<Node>, fromBlock: SugarElement<Element>, toBlock: SugarElement<Element>): Optional<CaretPosition> => {
   if (Empty.isEmpty(toBlock)) {
+    const descendants = SelectorFilter.descendants(toBlock, '*');
     Remove.remove(toBlock);
     if (Empty.isEmpty(fromBlock)) {
       PaddingBr.fillWithPaddingBr(fromBlock);
+      SelectorFind.child(fromBlock, '[data-mce-bogus]').fold(
+        Fun.noop,
+        (bogus) => Arr.each(descendants, (descendant) => Insert.wrap(bogus, descendant))
```

Comment:
This probably isn't doing what you think it's doing, either that or it's a weird way to do it, I'd like to know what you're thinking here.

`Insert.wrap` starts by inserting the first argument into the parent of the second argument, which will move `bogus` out of `fromBlock` and put it into `toBlock`, but `toBlock` has been removed from the DOM. I'm not sure how your test is passing. Even if it works in some scenarios this seems like it will lead to weird edge cases.

`InsertAll.append(bogus, descendants)` is a better way to append an array of elements into an element, and then if you want the `bogus` element to move please do so more clearly.

Reply:
I would expect this to be an Arr.foldl where it wraps on each iteration and returns the newly wrapped element for the next iteration.

Reply:
`InsertAll.append(bogus, descendants)` is not exactly what I want. The idea is to wrap the `bogus` with `toBlock` descendants. 

Update the code with more intuitive solution by @spocke. using `Arr.foldr` and returning the newly wrapped element.

https://github.com/tinymce/tinymce/blob/5d854d3e550ce06fca195f6cac6b2f3ce42d4b40/modules/tinymce/src/core/main/ts/delete/MergeBlocks.ts#L68-L72